dataprofiler/data_readers/data_utils.py

```diff
@@ -303,6 +302,43 @@ def detect_file_encoding(file_path, buffer_size=1024, max_lines=20):
     # Typical file representation is utf-8 instead of ascii, treat as such.
     if not encoding or encoding == 'ascii':
         encoding = 'utf-8'
+
+    # Check if encoding can be used to decode without throwing an error
+    def _decode_is_valid(encoding):
+        try: 
+            open(file_path, "rb").read().decode(encoding.lower())
+            return True
+        except: return False
```

Comment:
@lettergram @JGSweets this is how I am checking to see if I should use CharsetNormalizer/use more bytes - does this solution look better?

Reply:
This will never fail, as far as I can gather. @JGSweets can you share a couple files we know fail (if you haven't already)?

Then, I think you're going to have to try probably just check the encoding to see if it's a known failure state. 

At the end of the day, you'll probably have to try loading it (see the section of the code which loads the files). If there's a failure at that point, reinitialize the process with this method or something to that effect.

Reply:
Additionally, I think we might want to consider either opening using `with` or saving the open so we can close the file when done rather than leaving open.

maybe
```python
bytes_to_read = 1024*1024  # 1MB
with open(file_path, 'r', encoding=encoding.lower()) as fp:
    fp.read(bytes_to_read)
```